src/pydap/parsers/__init__.py

```diff
@@ -110,16 +108,21 @@ def parse_ce(query_string):
     Returns a tuple with the projection and the selection.
 
     """
-    tokens = [token for token in unquote(query_string).split("&") if token]
-    if not tokens:
+    if protocol == "dap2":
```

Comment:
This parsing of projections and selections only work for DAP2, since there is an ampersan there, but this part is only applicable for Sequences and Structures. A different PR will fix this to allow this behavior on DAP4 for Seqs and Structs